src/autify/mobile/mobilelink/mobile-link-manager/MobileLinkManager.ts

```diff
@@ -0,0 +1,273 @@
+    );
+    return processExit.code;
+  }
+
+  public async exit(options?: {
+    ignoreError: boolean;
+  }): Promise<number | null> {
+    try {
+      this.logger.debug("exit");
+      const snapshot = this.service.getSnapshot();
+      if (snapshot.done) {
+        this.logger.debug("Already exited.");
+        return snapshot.context.processExit?.code || null;
+      }
+
+      this.service.send("TERMINATE");
+      return await this.onceDone();
+    } catch (error) {
+      if (options?.ignoreError) {
+        this.logger.warn(`Ignoring exit error: ${error}`);
+        return null;
+      }
+
+      throw error;
+    }
+  }
+
+  private readonly clientEvents = new EventEmitter() as MobileLinkEvents;
+  private readonly logger: Logger;
+  private readonly clientLogger: Logger;
+  private readonly service: MobileLinkStateMachineService;
+  private childProcess?: ChildProcessWithoutNullStreams;
+
+  public constructor(private readonly options: MobileLinkManagerOptions) {
+    const level = "info";
+    this.logger = createManagerLogger({ level });
+    this.clientLogger = createMobileLinkLogger({ level });
+    this.service = createService({
+      start: (workspaceId) => this.linkStart(workspaceId),
+      terminate: () => this.terminate(),
+      kill: () => this.kill(),
+      cleanup: () => this.cleanup(),
+      errors: [],
+    }).onTransition((state, event) => {
+      if (!state.changed) return;
+      this.logger.debug(
+        `Transition to: ${state.value}, event: ${JSON.stringify(event)}`
+      );
+    });
+    process.on("SIGINT", () => this.service.send("TERMINATE"));
+    process.on("SIGTERM", () => this.service.send("TERMINATE"));
+    this.clientEvents.on("ready", () => {
+      this.service.send("READY");
+    });
+    this.clientEvents.on("log", (msg) => {
+      // Guardrail in case debug server isn't working well.
+      if (
+        msg.includes("Local devices are ready for use") ||
+        msg.includes(
+          "起動しました。ローカルデバイスを利用する準備ができました。"
+        )
+      ) {
+        this.clientEvents.emit("ready", msg);
+      }
+    });
+    this.clientEvents.on("error", (error) => {
+      this.logger.warn(`Ignoring client error: ${error}`);
+    });
+  }
+
+  private spawn(
+    argv: string[],
+    extraEnv: { [key: string]: string } = {},
+    connectConsole: boolean = true
+  ) {
+    this.childProcess = spawn(this.mobileLinkPath, argv, {
+      env: {
+        ...env,
+        ...extraEnv,
+      },
+    });
+    if (connectConsole) {
+      this.childProcess.stdout.on("data", (data) => {
+        process.stdout.write(data);
+      });
+
+      this.childProcess.stderr.on("data", (data) => {
+        process.stderr.write(data);
+      });
+    }
+  }
+
+  private linkStart(workspaceId?: string) {
+    try {
+      const argv = ["link", "start"];
+      if (workspaceId) argv.push(workspaceId);
+      this.spawn(
+        argv,
+        {
+          AUTIFY_CONNECT_BIN_PATH: this.autifyConnectPath,
```

Comment:
Don't we need to pass AUTIFY_MOBILE_ACCESS_TOKEN? If we want to align with current CLI usage.
It's unnecessary if we expect customers to run link setup without AUTIFY_MOBILE_ACCESS_TOKEN first.

Reply:
@rerorero 
Oh, that's true. Let me address it later